addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/handler/ValloxMVWebSocket.java

```diff
@@ -0,0 +1,357 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.valloxmv.handler;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.valloxmv.ValloxMVBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ValloxMVWebSocket} is responsible for socket communication with the vallox ventilation unit
+ *
+ * @author Björn Brings - Initial contribution
+ */
+public class ValloxMVWebSocket {
+    private String ip;
+    private ValloxMVHandler voHandler;
+    private Logger logger;
+    private WebSocketClient client;
+    private ValloxMVWebSocketListener socket;
+
+    public ValloxMVWebSocket(ValloxMVHandler voHandler, String ip) {
+        this.voHandler = voHandler;
+        this.ip = ip;
+        logger = LoggerFactory.getLogger(ValloxMVWebSocket.class);
+        client = new WebSocketClient();
+    }
+
+    public void request(ChannelUID channelUID, String updateState) {
+        try {
+            socket = new ValloxMVWebSocketListener(channelUID, updateState);
+            client.start();
+
+            URI destUri = new URI("ws://" + ip + ":80");
+
+            ClientUpgradeRequest request = new ClientUpgradeRequest();
+            logger.debug("Connecting to: {}", destUri);
+            client.connect(socket, destUri, request);
+            socket.awaitClose(10, TimeUnit.SECONDS);
+        } catch (IOException e) {
+            logger.error("Error: {}", e.getMessage().toString());
+        } catch (Exception e) {
+            logger.error("Error: {}", e.getMessage().toString());
```

Comment:
I got NPEs here. Please use something like this: `logger.warn("Error connecting vallox unit.", e);`